router/core/src/main/scala/io/buoyant/router/Router.scala

```diff
@@ -253,12 +253,17 @@ object StackRouter {
      * Install the TlsClientPrep module below the endpoint stack so that it
      * may avail itself of any and all params to set TLS params.
      */
-    def mkStack[Req, Rsp](orig: Stack[ServiceFactory[Req, Rsp]]): Stack[ServiceFactory[Req, Rsp]] =
-      orig ++ (TlsClientPrep.nop[Req, Rsp] +: stack.nilStack)
+    def mkStack[Req, Rsp](orig: Stack[ServiceFactory[Req, Rsp]]): Stack[ServiceFactory[Req, Rsp]] = {
+      val stk = new StackBuilder[ServiceFactory[Req, Rsp]](orig)
+      stk.push(TlsClientPrep.nop[Req, Rsp])
+      stk.result.remove(Retries.Role)
+    }
   }
 
   def newPathStack[Req, Rsp]: Stack[ServiceFactory[Req, Rsp]] = {
     val stk = new StackBuilder[ServiceFactory[Req, Rsp]](stack.nilStack)
+    stk.push(StatsFilter.module)
```

Comment:
It uses the stats receiver for the path stack which is scoped by `rt/<rt>/dst/path/<name>`.  The HttpEndToEndTest illustrates this.